base/src/com/google/idea/blaze/base/model/primitives/WorkspacePath.java

```diff
@@ -48,6 +48,7 @@ public static WorkspacePath createIfValid(String relativePath) {
    * @throws IllegalArgumentException if the path is invalid
    */
   public WorkspacePath(String relativePath) {
```

Comment:
Drive by: the comment above says:

> relative path that must use the Blaze specific separator char to separate

So is this not a case of the user/caller of this function using it incorrectly?

Reply:
I was just importing a bazel project using the BUILD files - that led to this error.

This creates the following `.bazelproject`

```
directories:
  main/java/example

targets:
  //main/java/example/...:all

additional_languages:
  # Uncomment any additional languages you want supported
  # android
  # dart
  # go
  # javascript
  # kotlin
  # scala
  # typescript
```

Reply:
It could be a case that the caller is using it incorrectly - I'll see if I can investigate the root at which the conversion takes place resulting in the windows line separators being introduced.

Reply:
When I get this (on windows) it's related to `Expand sync to working set`

Reply:
There are a lot of places where we don't correctly convert to canonical path formats -- we really don't support Windows at all yet.

We can't make this change as is in WorkspacePath, because it's used for both bazel target labels and files/directories. The proper fix is to audit every call site (e.g. GitWorkingSetProvider is one example which you'll hit with git projects and 'Expand sync to working set' active).

I would be open to making this change for Windows only though (or more generally, whenever File.separatorChar == '\\'), since it's an already-broken use case. Eventually we'd want to remove it though, and fix up the callers instead.